sqlmesh/core/engine_adapter/base.py

```diff
@@ -1002,34 +1016,53 @@ def create_schema(
         schema_name: SchemaName,
         ignore_if_exists: bool = True,
         warn_on_error: bool = True,
+        properties: t.List[exp.Expression] = [],
+    ) -> None:
+        return self._create_schema(
+            schema_name=schema_name,
+            ignore_if_exists=ignore_if_exists,
+            warn_on_error=warn_on_error,
+            properties=properties,
+            kind="SCHEMA",
+        )
+
+    def _create_schema(
+        self,
+        schema_name: SchemaName,
+        ignore_if_exists: bool,
+        warn_on_error: bool,
+        properties: t.List[exp.Expression],
+        kind: str,
     ) -> None:
         """Create a schema from a name or qualified table name."""
         try:
             self.execute(
                 exp.Create(
                     this=to_schema(schema_name),
-                    kind="SCHEMA",
+                    kind=kind,
                     exists=ignore_if_exists,
+                    properties=exp.Properties(  # this renders as '' (empty string) if expressions is empty
+                        expressions=properties
+                    ),
                 )
             )
         except Exception as e:
             if not warn_on_error:
                 raise
-            logger.warning("Failed to create schema '%s': %s", schema_name, e)
+            logger.warning("Failed to create %s '%s': %s", kind.lower(), schema_name, e)
 
     def drop_schema(
         self,
         schema_name: SchemaName,
         ignore_if_not_exists: bool = True,
         cascade: bool = False,
+        **drop_args: t.Dict[str, exp.Expression],
     ) -> None:
-        self.execute(
-            exp.Drop(
-                this=to_schema(schema_name),
-                kind="SCHEMA",
-                exists=ignore_if_not_exists,
-                cascade=cascade,
-            )
+        return self._drop_object(
+            name=schema_name,
+            exists=ignore_if_not_exists,
+            kind="SCHEMA",
+            cascade=cascade,
         )
```

Comment:
Should we propagate `drop_args` to `_drop_object` here? Or was it left out intentionally?

Reply:
good catch